File: src/main/java/org/cerion/stocklist/web/DataAPI.java
package org.cerion.stocklist.web;


import org.cerion.stocklist.Price;
import org.cerion.stocklist.PriceList;
import org.cerion.stocklist.model.Dividend;
import org.cerion.stocklist.model.Interval;
import org.cerion.stocklist.model.Quote;
import org.cerion.stocklist.model.Symbol;

import java.util.Date;
import java.util.List;
import java.util.Map;
import java.util.Set;

public interface DataAPI {

    @Deprecated
    PriceList getPrices(String symbol, Interval interval, int max); // TODO possibly replace with date version

    List<Price> getPrices(String symbol, Interval interval, Date start) throws Exception;

    List<Dividend> getDividends(String symbol);

    List<Symbol> getSymbols(Set<String> symbols);
    Symbol getSymbol(String symbol);

    Map<String,Quote> getQuotes(Set<String> symbols);
    Quote getQuote(String symbol);
}

File: src/main/kotlin/web/DataAPI.kt
package org.cerion.stocks.core.web


import org.cerion.stocks.core.PriceList
import org.cerion.stocks.core.PriceRow
import org.cerion.stocks.core.model.Dividend
import org.cerion.stocks.core.model.Interval
import org.cerion.stocks.core.model.Quote
import org.cerion.stocks.core.model.Symbol
import java.util.*

// TODO split this into multiple interfaces since nothing implements everything directly
interface DataAPI {

    fun getPriceList(symbol: String, interval: Interval, start: Date): PriceList

    @Deprecated("Use pricelist version")
    fun getPrices(symbol: String, interval: Interval, start: Date): List<PriceRow>

    fun getDividends(symbol: String): List<Dividend>

    fun getSymbols(symbols: Set<String>): List<Symbol>
    fun getSymbol(symbol: String): Symbol?

    fun getQuotes(symbols: Set<String>): Map<String, Quote>
    fun getQuote(symbol: String): Quote?
}
